### PR TITLE
supervisor-page

### PR DESCRIPTION
**Pull request by Shaden-mdn**
my work

### CODE REVIEW

**Comment by Shaden-mdn** [comment]
my work

File: app/Http/Controllers/SupervisorController.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\Supervisor;
+use App\Models\Leadre;
+use App\Models\LeaderDuty;
+
+class SupervisorController extends Controller
```
Comment (saraIsmailSE) [inline app/Http/Controllers/SupervisorController.php:10]: Duplicated?!

File: app/Models/LeaderDuty.php [hunk 2]
```diff
@@ -2,10 +2,17 @@
...
 use Illuminate\Database\Eloquent\Model;
 
 class LeaderDuty extends Model
 {
     use HasFactory;
+    protected $fillable = ['leader_id', 'week_id', 'follow_up_post', 'support_post', 'news', 'elementary_mark', 'final_mark',
+    'audit_final_mark', 'withdrawn_ambassadors', 'leader_reading', 'team_final_mark', 'current_team_members'];
+
+    public function leaders() {
+        return $this->belongsTo(Leadre::class, 'id');
```
Comment (saraIsmailSE) [inline app/Models/LeaderDuty.php:16]: Belongs by leader_id

File: app/Models/Leadre.php [hunk 3]
```diff
@@ -2,10 +2,21 @@
...
 {
     use HasFactory;
+    protected $fillable = ['name', 'week_id', 'advisor_id', 'supervisor_id', 'team', 'type'];
+
+    public function supervisors() {
+        return $this->belongsTo(Supervisor::class, 'user_id');
+    }
+
+    public function leaderDuties() {
+        return $this->hasOne(LeaderDuty::class, 'leader_id');
```
Comment (saraIsmailSE) [inline app/Models/Leadre.php:20]: Has many

File: database/migrations/2022_02_20_155407_create_supervisors_table.php [hunk 4]
```diff
@@ -16,7 +16,7 @@ public function up()
         Schema::create('supervisors', function (Blueprint $table) {
             $table->id();
             $table->integer('user_id');
-            $table->integer('current_advisor');
+            $table->integer('current_advisor')->nullable();
```
Comment (saraIsmailSE) [inline database/migrations/2022_02_20_155407_create_supervisors_table.php:19]: Same for previous advisor and team, they should be nullable

File: resources/views/Supervisor/supervisor.blade.php [hunk 5]
```diff
@@ -15,22 +15,22 @@
...
               <div class="page-title">
               <div class="row">
-               
-              <div class="col-10"> <h3>  الإضافات على عمل المراقب : </h3> 
+
+              <div class="col-10"> <h3>  الإضافات على عمل المراقب : </h3>
               <br>
 
-              <div class="alert alert-light-danger color-danger"><i class="bi bi-exclamation-circle"></i> 
+              <div class="alert alert-light-danger color-danger"><i class="bi bi-exclamation-circle"></i>
```
Comment (saraIsmailSE) [inline resources/views/Supervisor/supervisor.blade.php:26]: Need explanation

File: resources/views/Supervisor/supervisorHome.blade.php [hunk 6]
```diff
@@ -22,7 +27,13 @@
                                     </div>
                                     <div class="col-md-8">
                                         <h6 class="text-muted font-semibold"> حالتك الان</h6>
-                                        <h6 class="font-extrabold mb-0">مكلف</h6>
+                                        <h6 class="font-extrabold mb-0">
+                                            @if ($sp->current_advisor == null)
```
Comment (saraIsmailSE) [inline resources/views/Supervisor/supervisorHome.blade.php:31]: Use is_null function, it’s more efficient

File: resources/views/Supervisor/supervisorHome.blade.php [hunk 7]
```diff
@@ -42,7 +53,7 @@
                                     </div>
                                     <div class="col-md-8">
                                         <h6 class="text-muted font-semibold">موجهك الحالي</h6>
-                                        <h6 class="font-extrabold mb-0">شادن</h6>
+                                        <h6 class="font-extrabold mb-0">">{{ $sp->current_advisor }}</h6>
```
Comment (saraIsmailSE) [inline resources/views/Supervisor/supervisorHome.blade.php:56]: You need to check if there’s a current advisor

File: resources/views/Supervisor/supervisorHome.blade.php [hunk 8]
```diff
@@ -61,7 +72,7 @@
                                     </div>
                                     <div class="col-md-8">
                                         <h6 class="text-muted font-semibold">موجهك السابق</h6>
-                                        <h6 class="font-extrabold mb-0">نصر الدين </h6>
+                                        <h6 class="font-extrabold mb-0">">{{ $sp->previous_advisor }}</h6>
```
Comment (saraIsmailSE) [inline resources/views/Supervisor/supervisorHome.blade.php:75]: Same as current advisor, check first

**Pull request closed without merging**